optimum/habana/transformers/models/starcoder2/modeling_starcoder2.py

```diff
@@ -114,10 +123,35 @@ def gaudi_starcoder2_attention_forward(
 
         attn_weights = attn_weights + attention_mask
 
-    # upcast attention to fp32
-    attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(query_states.dtype)
-    attn_weights = nn.functional.dropout(attn_weights, p=self.attention_dropout, training=self.training)
-    attn_output = torch.matmul(attn_weights, value_states)
+    query_length = q_len if past_key_value is None else q_len + past_key_value.key_cache[self.layer_idx].shape[2]
+    # Taken from mpt PR 1101
+    if use_flash_attention and FusedSDPA:
+        import habana_frameworks.torch.hpu as ht
+
+        if query_length == 1:
+            # next token
+            with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
+                attn_output = FusedSDPA.apply(
+                    query_states, key_states, value_states, attention_mask, 0.0, False, None
+                )
+        else:
+            # first token
+            with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
+                if query_length > 8192:
```

Comment:
Can you verify/explain the 8192 value? It seems its the context window for starcoder and mpt but for starcoder2 the value seems to be 16,384

Reply:
Yeah an oversight, should be 16k. Added.